### PR TITLE
Updated CSS property autocompletion and other misc changes

### PR DESCRIPTION
**Pull request by Mitchman215**
@TheRealSyler 
Addresses #98.
I added a package script `update-css` that pulls css data from [@vscode/web-custom-data](https://www.npmjs.com/package/@vscode/web-custom-data), so hopefully this will make it easy to keep the data up to date. Currently, it only uses the properties data, but it also pulls in pseudo-elements/classes and at-rules, so that could be used in the future too. I also added a `useNewData` field to the "autcomplete.utility.ts" file that can revert to using the old css data if you want.
One last thing: I updated the dependencies in order to get the project to run for me and I added eslint for some linting. If you want me to change anything, just let me know.

### CODE REVIEW

**Comment by Mitchman215** [comment]
Sorry this took a while! I have been preoccupied with other things and am still getting used to typescript and vscode's API. That being said, I enjoyed working on this and would be interested in working more on this project. I have a few ideas that I can talk to you about if you are interested (I'm not sure how much you are still working on this project).
I noticed the "sass-lsp" branch, is that a language server implementation?

File: src/autocomplete/cssLanguageTypes.ts [hunk 1]
```diff
@@ -0,0 +1,313 @@
+// /*---------------------------------------------------------------------------------------------
```
Comment (TheRealSyler) [inline src/autocomplete/cssLanguageTypes.ts:1]: Im not sure what this file is supposed to do, i think you can just delete it.

**Review by TheRealSyler — CHANGES_REQUESTED** [review]
Looks good, just address the comments i made and if you decide to remove the `useNewData` variable without replacing it by an extension setting also remove the old code and the old data that is not being used.

Other than that the pr should be ready to merge.

**Comment by TheRealSyler** [comment]
no worries, we all have stuff to do, you have all the time in the world.

feel free to add any feature you want, if i don't think it fits the extension you can just copy and paste it into a new extension like i did with my [sass-next](https://github.com/TheRealSyler/vscode-sass-next) extension.

im no longer working on the sass-lsp branch(or any new features) and i don't remember what i wanted to do with it.

thanks for taking the time to contribute.

**Comment by Mitchman215** [comment]
@TheRealSyler 
Thanks for the feedback, I fixed all the things you made comments about.
I also made autocomplete use the new pseudo class/elements data.
Let me know if theres anything else I need to address before this is merged.

File: src/autocomplete/autocomplete.ts [hunk 2]
```diff
@@ -165,6 +165,7 @@ class SassCompletion implements CompletionItemProvider {
       );
     }
 
+    // console.log({ completions });
```
Comment (TheRealSyler) [inline src/autocomplete/autocomplete.ts:168]:
```suggestion
```

File: src/autocomplete/autocomplete.utility.ts [hunk 3]
```diff
@@ -42,76 +41,127 @@ export interface ImportsItem {
...
+      const completionItem = new CompletionItem(rawProp.name);
+      completionItem.detail = AutocompleteUtils.convertOptionalMarkup(rawProp.description);
+      completionItem.tags =
+        rawProp.status === 'obsolete' ? [CompletionItemTag.Deprecated] : [];
+      completionItem.sortText = '5';
+
+      completionItem.insertText = rawProp.name.concat(': ');
+      completionItem.kind = CompletionItemKind.Property;
+      completionItem.documentation = new MarkdownString(this.constructDocumentation(rawProp));
+      // TODO: ideally, trigger intellisense to autosuggest possible values of property
```
Comment (TheRealSyler) [inline src/autocomplete/autocomplete.utility.ts:158]:
```suggestion
```

File: src/autocomplete/autocomplete.utility.ts [hunk 4]
```diff
@@ -42,76 +41,127 @@ export interface ImportsItem {
...
+      + formattedRefs
+      + formattedValues
+    );
+
+    return documentation;
+  }
+
+  /** Constructs the snippet string to be used as the CompletionItem's insertText
+   *  for entities that could be functions.
+   *  Assumes that the parentheses are in the names if the enttity is indeed a function.
```
Comment (TheRealSyler) [inline src/autocomplete/autocomplete.utility.ts:126]:
```suggestion
   *  Assumes that the parentheses are in the names if the entity is indeed a function.
```

File: src/autocomplete/autocomplete.utility.ts [hunk 5]
```diff
@@ -42,76 +41,127 @@ export interface ImportsItem {
...
   static getPropertyName(currentWord: string): string {
     return currentWord.trim().replace(':', ' ').split(' ')[0];
   }
 
   /** Search for property in cssSchema */
-  static findPropertySchema(property: string): BasicRawCompletion {
-    return generatedPropertyData[property];
+  static findPropertySchema(property: string): IPropertyData {
+    return cssData.properties[property];
   }
```
Comment (TheRealSyler) [inline src/autocomplete/autocomplete.utility.ts:57]:
```suggestion
  static findPropertySchema(property: string) {
    return cssData.properties.find(prop => prop.name === property);
  }
```

File: src/scripts/generateCssData.ts [hunk 6]
```diff
@@ -23,12 +25,12 @@ const output = [
 	' *--------------------------------------------------------------------------------------------*/',
 	'// file generated from @vscode/web-custom-data NPM package',
 	'',
-	`import { ${DATA_TYPE} } from '../cssLanguageTypes';`,
+	`import { CSSDataV1 } from '../../autocomplete.interfaces';`,
```
Comment (TheRealSyler) [inline src/scripts/generateCssData.ts:28]:
```suggestion
	`import { ${DATA_TYPE} } from '../../autocomplete.interfaces';`,
```

File: tsconfig.json [hunk 7]
```diff
@@ -3,12 +3,13 @@
     "module": "commonjs",
     "target": "es6",
     "outDir": "out",
-    "lib": ["es6"],
+    "lib": ["es6", "ES2018"],  // ES2018 needed for regex groups support
```
Comment (TheRealSyler) [inline tsconfig.json:6]:
```suggestion
    "lib": ["es6", "ES2018"],
```

**Review by TheRealSyler — CHANGES_REQUESTED** [review]
There are a few small changes, i think either you or me can commit them directly via github, not sure how the feature works i never used it, if i can do it ill just merge the pr, thanks again for taking the time to contribute.

File: src/autocomplete/autocomplete.utility.ts (hunk 5 above)
Comment (TheRealSyler) [inline src/autocomplete/autocomplete.utility.ts:57]:
```suggestion
  static findPropertySchema(property: string) {
    return cssData.properties.find(prop => prop.name === property);
  }
```

**Pull request merged**